nowcasting_api/cache.py

```diff
@@ -1,208 +1,90 @@
-""" Caching utils for api"""
+"""Caching utils for api using fastapi-cache"""
 
-import json
 import os
-import time
-from datetime import datetime, timedelta, timezone
-from functools import wraps
+from typing import Any, Callable, Optional
 
 import structlog
-from database import save_api_call_to_db
+from fastapi import Request
+from fastapi_cache import FastAPICache
+from fastapi_cache.backends.inmemory import InMemoryBackend
+from fastapi_cache.decorator import cache
+
+from nowcasting_api.database import save_api_call_to_db
 
 logger = structlog.stdlib.get_logger()
 
-CACHE_TIME_SECONDS = 120
-cache_time_seconds = int(os.getenv("CACHE_TIME_SECONDS", CACHE_TIME_SECONDS))
-DELETE_CACHE_TIME_SECONDS = 240
-delete_cache_time_seconds = int(os.getenv("DELETE_CACHE_TIME_SECONDS", DELETE_CACHE_TIME_SECONDS))
+# Configuration constants with environment variable fallbacks
+CACHE_TIME_SECONDS = int(os.getenv("CACHE_TIME_SECONDS", 120))
+DELETE_CACHE_TIME_SECONDS = int(os.getenv("DELETE_CACHE_TIME_SECONDS", 240))
```

Comment:
do these get used anywhere?

Reply:
I have implemented a clear_cache_key function that utilizes the expiration parameter, and I modified the generate_cache_key function to check for these lock entries.